File: net/JetBrains.Serialization/src/Dataholders/SignatureTimeStampAttributeInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Attribute = Org.BouncyCastle.Asn1.Cms.Attribute;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.Fields)]
public class SignatureTimeStampAttributeInfo : AttributeInfo
{
  public override TextualInfo Identifier { get; }
  public List<RSASignedDataInfo> Content { get; }

  public SignatureTimeStampAttributeInfo(TextualInfo identifier, List<RSASignedDataInfo> content)
  {
    Identifier = identifier;
    Content = content;
  }

  public SignatureTimeStampAttributeInfo(Attribute attribute)
    : this(TextualInfo.GetInstance(attribute.AttrType),
      attribute.AttrValues.ToArray().Select(av => RSASignedDataInfo.GetInstance(av as DerSequence)).ToList())
  {
  }

  public override Asn1Encodable GetPrimitiveContent() =>
    Content.Cast<IEncodableInfo>().ToList().ToPrimitiveList().ToDerSet();
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/attributeInfos/SignatureTimeStampAttributeInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import com.jetbrains.signatureverifier.serialization.toPrimitiveDLSet
import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.DLSequence
import org.bouncycastle.asn1.cms.Attribute

@Serializable
data class SignatureTimeStampAttributeInfo(
  override val identifier: TextualInfo,
  val content: List<RSASignedDataInfo>
) : AttributeInfo() {

  constructor(attribute: Attribute) : this(
    TextualInfo.getInstance(attribute.attrType),
    attribute.attrValues.map { RSASignedDataInfo.getInstance(it as DLSequence) }
  )

  override fun getPrimitiveContent() = content.toPrimitiveDLSet()
}